src/backend/apps/feed/client.py

```diff
@@ -427,6 +427,15 @@ def get_high_elevation_forecast_list(self):
                         continue  # empty response, continue with next entry
                     data = response.json()
 
+                    warnings = data.get("Warnings") or {}
+                    if warnings.get("Events"):
+                        # Filter out any events with Type "ended"
+                        warnings["Events"] = [event for event in warnings["Events"] if event.get("Type") != "ended"]
+
+                    if not warnings["Events"]:
+                        warnings = None
+
```

Comment:
There are two distinct branches here for one condition, so this should be an if else, rather than sequential ifs.  Also, line 435: if there's no "Events" key, a KeyError exception will be raised.

Suggest replacing 435 with `else:`

Reply:
If the reason for sequential ifs is to set warnings to None in case there are no warnings (i.e., if the result of 433 is an empty list), then 435 will not work as intended: an empty list is a truthy value.  You would need to check that the length of the list is zero.

Reply:
I changed it to:
                        if len(warnings["Events"]) == 0: 
                            warnings = None
                    else:
                        warnings = None